usr/src/cmd/smbsrv/smbd/smbd_logon.c

```diff
@@ -193,7 +201,18 @@ smbd_user_auth_logon(smb_logon_t *user_info)
 		token->tkn_audit_sid = entry->sa_audit_sid;
 	}
 
+	if (buf != NULL)
```

Comment:
This test is unnecessary -- free(NULL) is safe in user space.

Reply:
Btw, it would be cool if adt_end_session() had the same safety (nop if argument is NULL).  Maybe it already does?  I haven't looked.

Reply:
Yes, adt_end_session is OK to call with NULL.

Reply:
Updated as suggested.